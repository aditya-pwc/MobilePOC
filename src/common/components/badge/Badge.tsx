import React, { FC, ReactNode } from 'react'
import { StyleSheet, View } from 'react-native'
import CText from '../CText'

type BadgeType = 'green' | 'yellow' | 'red' | 'grey' | 'green_outline' | 'grey_outline'

interface BadgeProps {
    type: BadgeType
    label: string
    leadingIcon?: ReactNode
}

const DEFAULT_CONFIG = {
    green: {
        containerStyle: { backgroundColor: '#2DD36F' },
        fontStyle: { color: '#FFFFFF' }
    },
    yellow: {
        containerStyle: { backgroundColor: '#FFC409' },
        fontStyle: { color: '#000000' }
    },
    red: {
        containerStyle: { backgroundColor: '#EB445A' },
        fontStyle: { color: '#FFFFFF' }
    },
    grey: {
        containerStyle: { backgroundColor: '#D3D3D3' },
        fontStyle: { color: '#565656' }
    },
    green_outline: {
        containerStyle: {
            backgroundColor: '#FFFFFF',
            borderWidth: 1,
            borderColor: '#565656'
        },
        fontStyle: { color: '#565656' }
    },
    grey_outline: {
        containerStyle: {
            backgroundColor: '#FFFFFF',
            borderWidth: 1,
            borderColor: '#2DD36F'
        },
        fontStyle: { color: '#2DD36F' }
    }
}

const styles = StyleSheet.create({
    containerStyle: {
        height: 20,
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'row',
        borderRadius: 10,
        paddingHorizontal: 7
    },
    labelStyle: {
        fontWeight: 'bold',
        fontSize: 12,
        textTransform: 'uppercase'
    }
})

export const Badge: FC<BadgeProps> = (props) => {
    const { type, label, leadingIcon } = props
    return (
        <View style={[styles.containerStyle, DEFAULT_CONFIG[type].containerStyle]}>
            {leadingIcon}
            <CText style={[styles.labelStyle, DEFAULT_CONFIG[type].fontStyle]}>{label}</CText>
        </View>
    )
}
