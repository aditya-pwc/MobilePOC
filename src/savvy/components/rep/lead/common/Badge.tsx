import React, { FC } from 'react'
import { StyleSheet, View } from 'react-native'
import CText from '../../../../../common/components/CText'

interface BadgeProps {
    label: string
}

const styles = StyleSheet.create({
    container: {
        marginRight: 10,
        backgroundColor: '#2DD36F',
        borderRadius: 20,
        paddingHorizontal: 10,
        paddingVertical: 3,
        marginTop: 10,
        alignItems: 'center',
        justifyContent: 'center'
    },
    textStyles: {
        color: 'white',
        fontWeight: '700'
    }
})

const Badge: FC<BadgeProps> = (props: BadgeProps) => {
    const { label } = props
    return (
        <View style={[styles.container, { width: label.length * 14 }]}>
            <CText style={styles.textStyles}>{label}</CText>
        </View>
    )
}

export default Badge
