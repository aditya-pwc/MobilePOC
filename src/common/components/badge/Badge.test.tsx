import React from 'react'
import renderer from 'react-test-renderer'
import { Badge } from './Badge'

describe('Badge', () => {
    it('green renders correctly', () => {
        const component = renderer.create(<Badge label={'test'} type={'green'} />)
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('yellow renders correctly', () => {
        const component = renderer.create(<Badge label={'test'} type={'yellow'} />)
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('red renders correctly', () => {
        const component = renderer.create(<Badge label={'test'} type={'red'} />)
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('grey renders correctly', () => {
        const component = renderer.create(<Badge label={'test'} type={'grey'} />)
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('green_outline renders correctly', () => {
        const component = renderer.create(<Badge label={'test'} type={'green_outline'} />)
        expect(component.toJSON()).toMatchSnapshot()
    })

    it('grey_outline renders correctly', () => {
        const component = renderer.create(<Badge label={'test'} type={'grey_outline'} />)
        expect(component.toJSON()).toMatchSnapshot()
    })
})
