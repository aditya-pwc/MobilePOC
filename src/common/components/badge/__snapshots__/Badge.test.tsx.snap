// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge green renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "center",
        "paddingHorizontal": 7,
      },
      Object {
        "backgroundColor": "#2DD36F",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontFamily": "Gotham-Book",
        },
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#FFFFFF",
          },
        ],
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`Badge green_outline renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "center",
        "paddingHorizontal": 7,
      },
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#565656",
        "borderWidth": 1,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontFamily": "Gotham-Book",
        },
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#565656",
          },
        ],
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`Badge grey renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "center",
        "paddingHorizontal": 7,
      },
      Object {
        "backgroundColor": "#D3D3D3",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontFamily": "Gotham-Book",
        },
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#565656",
          },
        ],
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`Badge grey_outline renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "center",
        "paddingHorizontal": 7,
      },
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#2DD36F",
        "borderWidth": 1,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontFamily": "Gotham-Book",
        },
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#2DD36F",
          },
        ],
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`Badge red renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "center",
        "paddingHorizontal": 7,
      },
      Object {
        "backgroundColor": "#EB445A",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontFamily": "Gotham-Book",
        },
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#FFFFFF",
          },
        ],
      ]
    }
  >
    test
  </Text>
</View>
`;

exports[`Badge yellow renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "center",
        "paddingHorizontal": 7,
      },
      Object {
        "backgroundColor": "#FFC409",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "fontFamily": "Gotham-Book",
        },
        Array [
          Object {
            "fontSize": 12,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          Object {
            "color": "#000000",
          },
        ],
      ]
    }
  >
    test
  </Text>
</View>
`;
